### PR TITLE
Newarmorstand

### PR DESCRIPTION
**Pull request by MiniDigger**
Adds a api for spawning armorstands models from a file. These files are generated using mrgarettos armorstan model generator.

### CODE REVIEW

**Comment by MiniDigger** [comment]
test

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModel.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * Spawns the model at the given location
+     *
+     * @param rootLocation the location to spawn the model
+     * @param plugin       the plugin that spawns the model
+     */
+    public void spawn( Location rootLocation, Plugin plugin ) {
+        entities.forEach( ( entity ) -> entity.spawn( rootLocation, plugin ) );
```
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModel.java]: No streams please

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModel.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+
+    /**
+     * Despawns this model
+     */
+    public void despawn() {
+        entities.forEach( ArmorStandModelEntity::despawn );
```
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModel.java]: No streams please

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelEntity.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                loc.setY( loc.getY() - 1.5 );
+                loc.setPitch( (float) headPose.getY() );
+                loc.setYaw( (float) headPose.getX() );
+
+                armorStand = spawnArmorStand( loc, plugin );
+                armorStand.setVisible( false );
+                armorStand.setAI( false );
+
+                villager = (Villager) loc.getWorld().spawnEntity( loc, EntityType.VILLAGER );
+                EntityVillager entityVillager = ( (CraftVillager) villager ).getHandle();
```
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelEntity.java:90]: Maybe a bit more abstraction into another module "nms" or something?

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelHandler.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ */
+public class ArmorStandModelHandler {
+
+    private List<ArmorStandModel> models = new ArrayList<>();
+
+    /**
+     * Called onDisable, will despawn all remaining models
+     */
+    public void disable() {
+        models.forEach( ArmorStandModel::despawn );
```
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelHandler.java]: No streams please

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelImporter.java [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+                        entity.setLocation( location );
+                        entites.add( entity );
+                    } catch ( Exception e ) {
+                        e.printStackTrace();
+                        log.warning( e.getMessage() );
+                        continue;
+                    }
+                }
+                //solid
+                else if ( command.startsWith( "setblock" ) ) {
```
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelImporter.java]: Syntax should be "} else if {"

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelImporter.java [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+                    if ( temp.length == 5 ) {
+                        item.setDurability( Short.parseShort( temp[4] ) );
+                    }
+
+                    ArmorStandModelEntity entity = new ArmorStandModelEntity( item, ArmorStandModelSize.SOLID, "", new Vector( 0, 0, 0 ) );
+                    entity.setLocation( location );
+                    entites.add( entity );
+                }
+                //unknown
+                else {
```
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelImporter.java]: Same here for "} else {"

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelSize.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package net.devintia.commons.bukkit.armorstand;
+
+/**
+ * The different sizes and thier values
+ *
+ * @author MiniDigger
+ * @version 1.0.0
+ */
+enum ArmorStandModelSize {
+    SMALL( 0.28125 ), MEDIUM( 0.4375 ), LARGE( 0.625 ), SOLID( 1 );
```
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelSize.java]: enums should have one entry at a line

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/CustomControllerLook.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    private EntityInsentient a;
+
+    CustomControllerLook( EntityInsentient entityInsentient ) {
+        super( entityInsentient );
+        this.a = entityInsentient;
+    }
+
+    @Override
+    public void a() {
+        //fucka you biatch
```
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/CustomControllerLook.java]: Please explain why you did empty the method

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelImporter.java (hunk 5 above)
Comment (MiniDigger) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/armorstand/ArmorStandModelImporter.java]: what about the comments then?

**Pull request merged**